src/contract.rs

```diff
@@ -353,10 +375,41 @@ fn compute_current_metrics(
         // This is the new Delegated amount after slashing Ex: (10 => 9.8 etc.,)
         let current_delegated_amount = delegation.amount.amount.clone();
 
+        let self_delegation_opt = deps.querier.query_delegation(
+            validator_addr.account_address.clone(), //This is the Account Address
+            validator_addr.operator_address.clone(), //This is the Operator Address
+        );
+
+        // This is the self_delegation amount (delegation by validator)
+        let self_delegation_amount = match self_delegation_opt {
+            Ok(delegation_result) => delegation_result.unwrap().amount.amount,
+            Err(_) => Uint128::new(0),
+        };
+
+        // this stores current_slashing pointer value
+        // if there are no prev metric then it's default value is 1.0
+        let mut current_slashing_pointer = Decimal::one();
+
+        // formula=(current_delegated_amount/prev_delegated_amount)*prev_slashing_pointer
+        if !current_metrics.is_empty() {
+            let most_recent_metric = current_metrics.last().unwrap();
+            current_slashing_pointer = decimal_division_in_256(
+                uint128_to_decimal(current_delegated_amount),
+                uint128_to_decimal(most_recent_metric.delegated_amount),
+            );
+            current_slashing_pointer = decimal_multiplication_in_256(
+                current_slashing_pointer,
+                most_recent_metric.slashing_pointer,
+            );
+        }
+
         current_metrics.push(ValidatorMetrics {
-            addr: validator_addr.clone(),
+            operator_addr: validator_addr.operator_address.clone(),
+            account_addr: validator_addr.account_address.clone(),
```

Comment:
probably don't need account & operator_addr in ValidatorMetrics. Do we?

Reply:
removed account_addr from Validator_Metrics. Fixed some lint errors and test suite errors.